astrocut/make_cube.py

```diff
@@ -346,3 +349,500 @@ def make_cube(self, file_list, cube_file="img-cube.fits", sector=None, max_memor
+class TicaCubeFactory():
+
+    def __init__(self, max_memory=50):
+        """ Setting up the class members. NOTE: Later down the line 
+        we might want to generalize CubeFactory so that we no longer need 
+        a separate class for the TICA cubes. 
+        
+        Modifying the __init__ 
+        in CubeFactory so that it can switch parameters based on whether 
+        we want cubes for TICA or TESS will probably be the route I take.
+        Or see how this is generalized when Zcut wraps CubeFactory? 
+        """
+
+        self.max_memory = max_memory  # in GB
+        self.block_size = None  # Number of rows
+        self.num_blocks = None
+        self.cube_shape = None
+        
+        self.time_keyword = 'STARTTJD'  # Start time in TJD. TICA-specific.
+        self.last_file_keywords = ['ENDTJD']  # Stop time in TJD. TICA-specific (assumed to be in extension 0)                  
+        self.image_header_keywords = ['CAMNUM', 'CCDNUM']  # Camera number and CCD number 
+        self.template_requirements = {'NAXIS': 2}  # Using NAXIS instead of WCSAXES. JENNY: Verify this is OK.
+                                                   
+        self.file_list = None
+        self.template_file = None
+        
+        self.primary_header = None
+        self.info_table = None
+        self.cube_file = None
+
+        # Used when updating an existing cube 
+        self.old_cols = None
+        self.update = False
+        self.cube_append = None
+
+    def _configure_cube(self, file_list, **extra_keywords):
+        """ Run through all the files and set up the  basic parameters for the cube.
+        Set up the cube primary header.
+        """
+
+        file_list = np.array(file_list)
+        image_shape = None
+        start_times = np.zeros(len(file_list))
+        for i, ffi in enumerate(file_list):
+            ffi_data = fits.open(ffi, mode='denywrite', memmap=True)
+            
+            start_times[i] = ffi_data[0].header.get(self.time_keyword)
+
+            if image_shape is None:  # Only need to fill this once
+                image_shape = ffi_data[0].data.shape
+            
+            if self.template_file is None:  # Only check this if we don't already have it
+
+                is_template = True
+                for key, value in self.template_requirements.items():
+                    if ffi_data[0].header.get(key) != value:  # Checking for a good image header
+                        is_template = False
+                        
+                if is_template:
+                    self.template_file = ffi
+                    
+            ffi_data.close()
+
+        self.file_list = file_list[np.argsort(start_times)]
+
+        # Working out the block size and number of blocks needed for writing the cube
+        # without using too much memory
+        slice_size = image_shape[1] * len(self.file_list) * 2 * 4  # in bytes (float32)
+        max_block_size = int((self.max_memory * 1e9)//slice_size)
```

Comment:
```suggestion
        max_block_size = (self.max_memory * 1e9) // slice_size
```

no need to call `int()` here. realize this was copy / pasted